ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -63,7 +65,20 @@ function(ament_target_dependencies target)
         # otherwise use the classic CMake variables
         list_append_unique(definitions ${${package_name}_DEFINITIONS})
         list_append_unique(include_dirs ${${package_name}_INCLUDE_DIRS})
-        list(APPEND libraries ${${package_name}_LIBRARIES})
+        if(NOT "${${package_name}_LIBRARY_DIRS}" STREQUAL "")
+          foreach(library ${${package_name}_LIBRARIES})
+            if (NOT IS_ABSOLUTE ${library} OR NOT EXISTS ${library})
+              find_library(lib NAMES ${library} PATHS ${${package_name}_LIBRARY_DIRS} NO_DEFAULT_PATH)
+              if(NOT lib)
+                message(FATAL_ERROR "ament_target_dependencies() ${library} library not found in ${package_name}_LIBRARY_DIRS")
+              endif()
+              set(library ${lib})
+            endif()
+            list(APPEND libraries ${library})
+          endforeach()
+        else()
+          list(APPEND libraries ${${package_name}_LIBRARIES})
```

Comment:
The logic could be modified to avoid the duplication of `list(APPEND ...)`. The `foreach` loop can always be used and in the loop `library` is conditionally updated before being appended.

Reply:
That'll force a check for `${package_name}_LIBRARY_DIRS` on every iteration.

Reply:
Ok, a simple string comparision. What is the concern with that?

Reply:
Do you think not duplicating that `list(APPEND ...)` call is worth adding that overhead to our builds? Every package uses `ament_target_dependencies()`, often multiple times. We'll exercise this code path a lot.

Reply:
I doubt you could even measure the runtime difference if that is what you are concerned about. Imo the code s much easier to read without the current conditional logic: loop over the libs, modify them conditionally and add them to a list.

Reply:
See https://github.com/ament/ament_cmake/pull/245/commits/711566d7c3ac4dc0952ffe9a3c0abb5dae84ac58.